### ISSUE TITLE
Webflow login does not work

### ISSUE THREAD

**Issue by tflidd** [nextcloud/desktop#730]
### Expected behaviour
Webflow procedure should work and lead to a synchronisation with the server.

### Actual behaviour
It does not work with TOTP, you get login field then you enter user/pw, then you send both. The
With TOTP, apache.log:

```
a.b.c.d - - [18/Oct/2018:09:12:39 +0000] "POST /nextcloud/login?redirect_url=/nextcloud/login/challenge/totp%3Fredirect_url%3D/nextcloud/login/challenge/totp%253Fredirect_url%253D/nextcloud/login/flow/grant%25253FclientIdentifier%25253D%252526stateToken%25253DV0lWpcOoOI1teV2r5t4xXRstZK9L2v8t9bt1zwl37SYGbd70i67xiboGP5TaiI1c HTTP/1.1" 303 1360 "-" "Mozilla/5.0 (Windows) mirall/2.5.0daily (build 20181017) (Nextcloud)"
a.b.c.d - - [18/Oct/2018:09:12:40 +0000] "GET /nextcloud/login/challenge/totp?redirect_url=/nextcloud/login/challenge/totp%3Fredirect_url%3D/nextcloud/login/challenge/totp%253Fredirect_url%253D/nextcloud/login/flow/grant%25253FclientIdentifier%25253D%252526stateToken%25253DV0lWpcOoOI1teV2r5t4xXRstZK9L2v8t9bt1zwl37SYGbd70i67xiboGP5TaiI1c HTTP/1.1" 303 1184 "-" "Mozilla/5.0 (Windows) mirall/2.5.0daily (build 20181017) (Nextcloud)"
a.b.c.d - - [18/Oct/2018:09:12:40 +0000] "GET /nextcloud/login?redirect_url=/nextcloud/login/challenge/totp%3Fredirect_url%3D/nextcloud/login/challenge/totp%253Fredirect_url%253D/nextcloud/login/challenge/totp%25253Fredirect_url%25253D/nextcloud/login/flow/grant%2525253FclientIdentifier%2525253D%25252526stateToken%2525253DV0lWpcOoOI1teV2r5t4xXRstZK9L2v8t9bt1zwl37SYGbd70i67xiboGP5TaiI1c HTTP/1.1" 200 3268 "-" "Mozilla/5.0 (Windows) mirall/2.5.0daily (build 20181017) (Nextcloud)"
a.b.c.d - - [18/Oct/2018:09:12:40 +0000] "GET /nextcloud/core/js/oc.js?v=29299b7c HTTP/1.1" 200 6989 "-" "Mozilla/5.0 (Windows) mirall/2.5.0daily (build 20181017) (Nextcloud)"
```

Then I disabled TOTP for this user, I managed to login but it is reloading all the time:

```
a.b.c.d - - [18/Oct/2018:09:27:38 +0000] "GET /nextcloud/apps/files/ HTTP/1.1" 200 8005 "-" "Mozilla/5.0 (Windows) mirall/2.5.0daily (build 20181017) (Nextcloud)"
a.b.c.d - - [18/Oct/2018:09:27:38 +0000] "GET /nextcloud/core/js/oc.js?v=29299b7c HTTP/1.1" 200 7504 "-" "Mozilla/5.0 (Windows) mirall/2.5.0daily (build 20181017) (Nextcloud)"
a.b.c.d - - [18/Oct/2018:09:27:39 +0000] "GET /nextcloud/apps/gallery/config?extramediatypes=1 HTTP/1.1" 412 1408 "-" "Mozilla/5.0 (Windows) mirall/2.5.0daily (build 20181017) (Nextcloud)"
a.b.c.d - - [18/Oct/2018:09:27:39 +0000] "GET /nextcloud/apps/files_sharing/api/externalShares HTTP/1.1" 401 1001 "-" "Mozilla/5.0 (Windows) mirall/2.5.0daily (build 20181017) (Nextcloud)"
a.b.c.d - - [18/Oct/2018:09:27:39 +0000] "PROPFIND /nextcloud/remote.php/dav/files/user/ HTTP/1.1" 401 1112 "-" "Mozilla/5.0 (Windows) mirall/2.5.0daily (build 20181017) (Nextcloud)"
a.b.c.d - - [18/Oct/2018:09:27:39 +0000] "GET /nextcloud/apps/systemtags/lastused HTTP/1.1" 401 1139 "-" "Mozilla/5.0 (Windows) mirall/2.5.0daily (build 20181017) (Nextcloud)"
a.b.c.d - - [18/Oct/2018:09:27:39 +0000] "GET /nextcloud/ocs/v2.php/core/whatsnew?format=json HTTP/1.1" 401 1059 "-" "Mozilla/5.0 (Windows) mirall/2.5.0daily (build 20181017) (Nextcloud)"
```

However, I still have a ownCloud client running that has no problem:

`a.b.c.d - user [18/Oct/2018:09:27:40 +0000] "PROPFIND /nextcloud/remote.php/dav/files/user/ HTTP/1.1" 207 4691 "-" "Mozilla/5.0 (Windows) mirall/2.5.0 (build 10560)"`

On the personal settings in my browser, I see that app passwords were created:
![image](https://user-images.githubusercontent.com/8111200/47148566-8dd68480-d2c0-11e8-850f-6f2ddfe40c1f.png)

Next idea, create an app password myself and use this for authentication. I get app password, I manage to find the right spot, I enter the credentials. If I hit "grant access" nothing happens. Apache access log shows no entries.

I installed different test versions in subfolders, perhaps there is something a bit different to the default default config. However, I have been using the ownCloud client for quite some time and never had problems. I haven't checked other clients/setups yet. However, I expected it to run smoothly. No idea how I could set this up.

### Client configuration
Client version: Daily build (https://download.nextcloud.com/desktop/daily/Windows/Nextcloud-2.5.0.61465-daily-20181017.exe)

Operating system: Win10

OS language: ENG

### Server configuration

Operating system: Debian

Web server: Apache 2

Database: MariaDB

PHP version: PHP 7.0

Nextcloud version: NC 14.0.3

Storage backend (external storage): no

**Comment by rullzer** [comment]
Could you provide me with test credentials?

**Comment by tflidd** [comment]
I sent you credentials to ro*@fam*.nl

**Comment by rullzer** [comment]
Thanks I'll try to take a look soon

**Comment by rullzer** [comment]
I just checked with current master (on linux) and I can authenticate without issues.

Could you make a quick screencast of what the client does for you?

**Comment by tflidd** [comment]
I installed a screencast software and now it works. I first tried a new user and after that the user that had problems last week.

What was different:
I had installed the Nextcloud client and just deinstalled the ownCloud client before. Second thing, I had the 2FA authentication active for that account and just disabled it. Now it has been disabled for a week. Not sure if any of this is cached somewhere and just needs some waiting time or system restart.

Now with the system working it is great. However, last week I felt pretty helpless. And I don't want to ask any user to change their client because in case of a problem, I have no way of helping and to tell them to wait for a week isn't so good either.
